### PR TITLE
[prometheus-metrics-adapter] Add logs and resync time x2

### PR DESCRIPTION
**Pull request by nabokihms**
Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

## Description
* Refactor reverse proxy code
* Make metrics resync interval to equal scrape interval x2

## Why do we need it, and what problem does it solve?

We see sometimes PMA returns not found to our queries. This is an attempt to fix the problem. PMA documentation quotation:

> My metrics appear and disappear:
You probably have a Prometheus collection interval or computation interval that's larger than your adapter's discovery interval.
The adapter only considers metrics with datapoints in the window [now-discoveryInterval, now] (in order to only capture metrics that are still present), so make sure that your discovery interval is at least as large as your collection interval.

In addition, this PR closes https://github.com/deckhouse/deckhouse/issues/1708. Logs example: 
```
R2022/07/07 17:39:50 6cbb5008-7da2-439f-8241-dffc9a091a00 -- Start request: 127.0.0.1:37914 [Go-http-client/1.1] GET https://trickster.d8-monitoring.svc.cluster.local/trickster/main/api/v1/query?query=sum%28irate%28container_cpu_usage_seconds_total%7Bnamespace%3D%22default%22%2Cpod%3D~%22pme-6cdcfbcc84-z8dgn%7Cvector-1654712930-0%22%2C+container%21%3D%22POD%22%7D%5B5m%5D%29%29+by+%28pod%2Ccontainer%29&time=1657215590.754
R2022/07/07 17:39:51 6cbb5008-7da2-439f-8241-dffc9a091a00 -- Response: 200, 644.065967ms
```

## What is the expected result?
PMA pods will be restarted.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests are passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: prometheus-metrics-adapter
type: feature
summary: Add logs and resync time x2
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/gcp'...

File: modules/301-prometheus-metrics-adapter/images/prometheus-reverse-proxy/app/server/logger.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+package server
+
+import (
+	"log"
+	"os"
+)
+
+var (
+	infLog = log.New(os.Stdout, "I", log.LstdFlags)
```
Comment (name212) [inline modules/301-prometheus-metrics-adapter/images/prometheus-reverse-proxy/app/server/logger.go:25]: I often use `kubectl logs blah-blah | grep -i error` for check troubles in log.

I think, better use prefix `ERROR` and `INFO`.

File: modules/301-prometheus-metrics-adapter/images/prometheus-reverse-proxy/app/server/server.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	return &Server{
+		PrometheusURL: promURL,
+		Transport:     transport,
+		Client:        httpClient,
+		ProxyPass:     proxy,
+	}
+}
+
+func (s *Server) Listen() {
+	listenAddr := "0.0.0.0:8000"
```
Comment (name212) [inline modules/301-prometheus-metrics-adapter/images/prometheus-reverse-proxy/app/server/server.go:79]: Maybe move into env variables?

File: modules/301-prometheus-metrics-adapter/images/prometheus-reverse-proxy/app/server/logger.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	errLog = log.New(os.Stderr, "E", log.LstdFlags)
```
Comment (name212) [inline modules/301-prometheus-metrics-adapter/images/prometheus-reverse-proxy/app/server/logger.go:26]: I often use `kubectl logs blah-blah | grep -i error` for find troubles. It is fast way. Find `E` letter in log is hard way.

I think we need use `error` and `info` here.

**Review by name212 — CHANGES_REQUESTED** [review]
https://github.com/deckhouse/deckhouse/pull/1970/files#r919864302
https://github.com/deckhouse/deckhouse/pull/1970/files#r919877118

File: modules/301-prometheus-metrics-adapter/images/prometheus-reverse-proxy/app/server/logger.go (hunk 3 above)
Comment (nabokihms) [reply modules/301-prometheus-metrics-adapter/images/prometheus-reverse-proxy/app/server/logger.go:26]:
```
kubectl logs blah-blah | grep -e '^E'
```

This format is similar to `klog`. Example:
``` 
I0713 09:51:26.480736       1 namespace_controller.go:185] Namespace has been deleted upmeter-control-plane-namespace-e430d6b3
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/azure'...

File: modules/301-prometheus-metrics-adapter/images/prometheus-reverse-proxy/app/server/logger.go (hunk 3 above)
Comment (name212) [reply modules/301-prometheus-metrics-adapter/images/prometheus-reverse-proxy/app/server/logger.go:26]: okay

Comment (name212) [reply modules/301-prometheus-metrics-adapter/images/prometheus-reverse-proxy/app/server/logger.go:26]: okay

**Review by name212 — APPROVED** [review]

**Pull request merged**